connectivity/check/policy.go

```diff
@@ -221,6 +222,31 @@ func authRequiredDropReason(flow *flowpb.Flow) bool {
 	return flow.GetDropReasonDesc() == flowpb.DropReason_AUTH_REQUIRED
 }
 
+// expectMetricsToIncrease compares metrics retrieved before any action were run and after;
+// may return an error if metrics did not increase.
+func expectMetricsToIncrease(metrics []string) metricsCompareFunc {
+	return func(before, after map[string]*dto.MetricFamily) error {
+		for _, metricName := range metrics {
+			bValue, ok := before[metricName]
+			if !ok {
+				// TODO DONIA: should I return an error here, should I store the error and range over all the metrics?
+				return fmt.Errorf("failed to retrieve metric %s in first retrieved metrics", metricName)
```

Comment:
Yeah, I think it's better to use `errors.Join` here to report on all metrics. It will make debugging test runs easier if you have more information on what worked and what didn't.

Reply:
Fixed, thanks!